nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessGroupResource.java

```diff
@@ -4597,6 +4611,246 @@ public Response deleteReplaceProcessGroupRequest(
+    })
+    public Response submitAnalyzeFlowRequest(
+        @ApiParam(
+            value = "The id of the process group representing (a part of) the flow to be analyzed.",
+            required = true
+        )
+        @PathParam("processGroupId")
+        final String processGroupId
+    ) {
+        if (isReplicateRequest()) {
+            return replicate(HttpMethod.POST);
+        }
+
+        NiFiUser user = NiFiUserUtils.getNiFiUser();
+
+        ProcessGroupEntity requestProcessGroupEntity = new ProcessGroupEntity();
+        requestProcessGroupEntity.setId(processGroupId);
+
+        return withWriteLock(
+            serviceFacade,
+            requestProcessGroupEntity,
+            lookup -> {
+                final ProcessGroupAuthorizable processGroup = lookup.getProcessGroup(processGroupId);
+                processGroup.getAuthorizable().authorize(authorizer, RequestAction.READ, user);
+            },
+            null,
+            (processGroupEntity) -> {
+                String analyzedGroupId = processGroupEntity.getId();
+
+                final String requestId = generateUuid();
+                final AsynchronousWebRequest<String, Void> analyzeFlowAsyncWebRequest = new StandardAsynchronousWebRequest<>(
+                        requestId,
+                        analyzedGroupId,
+                        analyzedGroupId,
+                        user,
+                        Collections.singletonList(new StandardUpdateStep("Analyze Process Group"))
+                );
+
+                // Submit the request to be performed in the background
+                final Consumer<AsynchronousWebRequest<String, Void>> analyzeFlowTask = asyncRequest -> {
+                    try {
+                        serviceFacade.analyzeProcessGroup(analyzedGroupId);
+                        asyncRequest.markStepComplete();
+                    } catch (final Exception e) {
+                        logger.error("Failed to run flow analysis on process group " + processGroupId, e);
+                        asyncRequest.fail("Failed to run flow analysis on process group " + processGroupId + " due to " + e);
+                    }
+                };
+                flowAnalysisAsyncRequestManager.submitRequest(
+                        FLOW_ANALYSIS_REQUEST_TYPE,
+                        requestId,
+                        analyzeFlowAsyncWebRequest,
+                        analyzeFlowTask
+                );
+
+                return generateOkResponse(createAnalyzeFlowRequestEntity(analyzeFlowAsyncWebRequest, requestId)).build();
+            }
+        );
+    }
+
+    /**
+     * Checks the status of an outstanding request for a flow analysis.
+     *
+     * @param requestId The id of flow analysis request
+     * @return An analyzeFlowRequestEntity
+     */
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("flow-analysis/{requestId}")
```

Comment:
This path overlaps with the `POST` request for Process Groups, perhaps changing the path to `flow-analysis-requests/{requestId}` would be better.
```suggestion
    @Path("{id}/flow-analysis-requests/{requestId}")
```

Reply:
The overlap is kind of intentional. These 3 methods are really the CURD (without U) of the flow analysis requests.
But I see the point that the POST needs a group id, the GET and the DELETE needs a request id.

So maybe the paths should be these instead:
```java
    @POST
    @Path("{id}/flow-analysis")

    @GET
    @Path("flow-analysis/{requestId}")

    @DELETE
    @Path("flow-analysis/{requestId}")
```

Reply:
That's a good point, but I believe the reason for including the Process Group ID in the path is for authorization handling, @mcgilman should be able to provide further clarification.